### ISSUE TITLE
Unintuitive adapter syntax

### ISSUE THREAD

**Issue by grzes314** [minerva-ml/steppy#28]
Current syntax for adapters has some peculiarities.
Consider the following example. 
```
        step = Step(
            name='ensembler',
            transformer=Dummy(),
            input_data=['input_1'],
            adapter={'X': [('input_1', 'features')]},
            cache_dirpath='.cache'
        )
```
This step basically extracts one element of the input. It seems redundant to write brackets and parentheses. Doing
```adapter={'X': ('input_1', 'features')},```
should be sufficient.

Moreover, to my suprise
```adapter={'X': [('input_1', 'features'), ('input_2', 'extra_features')]},```
is incorrect, and currently leads to
```ValueError: too many values to unpack (expected 2)```

My suggestions to make the syntax consistent are:
1. `adapter={'X': ('input_1', 'features')}` should map `X` to extracted `features`.
2. `adapter={'X': [...]}` should map `X` to a list of extracted objects (specified by elements of the list). In particular `adapter={'X': [('input_1', 'features')]}` should map `X` to a one-element list with extracted `features`.
3. `adapter={'X': ([...], func)}` should extract appropriate objects and put them on the list, then `func` should be called on that list, and `X` should map to the result of that call.

**Comment by jakubczakon** [comment]
@grzes314 great idea, it seems intuitive to have it as you described. 
Current api is (especially for just one element) very clunky. Also I think one option that could be useful is to be able to pass a chain of functions simply for example np.exp(lambda x: x[0]) or something.

**Comment by kamil-kaczmarek** [comment]
merged in #42. Closing this.
